Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,213 @@
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <string>
+#include <iostream>
+#include <vector>
+#include <boost/filesystem.hpp>
+#include "MolSupplier.h"
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+namespace fs = boost::filesystem;
+
+namespace RDKit{
+	struct SupplierOption{
+		bool takeOwnership = true;
+		bool sanitize = true;
+		bool removeHs = true;
+		bool strictParsing = true;
+
+		std::string delimiter = "\t";
+		int smilesColumn = 0;
+		int nameColumn = 1;
+		bool titleLine = true;
+
+		std::string nameRecord = "";
+		int confId2D = -1;
+		int confId3D = 0;
+	};
+
+  class GeneralFileReader{
+	  public:
+			std::string fname;
+		  std::string file_format, compression_format;
+
+		struct SupplierOption opt;	// options for the Mol Supplier 
+		public:
+			GeneralFileReader(std::string fileName);
+		  GeneralFileReader(std::string fileName, struct SupplierOption options);
+
+		  /*
+		    Function to check the validity of the file and compression format
+	    */
+		  bool valid();
+
+		  /*
+		    Function to set the file and compression format
+		  */
+		  void setFormat();
+
+		  /*
+		    Get MolSupplier Object based on the file and compression format
+	    */
+		  RDKit::MolSupplier* getSupplier();
+  };
+
+	GeneralFileReader::GeneralFileReader(std::string fileName){
+		fname = fileName;
+		file_format = "";
+		compression_format = "";
+	}
+
+	GeneralFileReader::GeneralFileReader(std::string fileName, struct SupplierOption options){
+		fname = fileName;
+		file_format = "";
+		compression_format = "";
+		opt = options;
+	}
+
+	bool GeneralFileReader::valid(){
+
+		std::vector<std::string > file_formats{ "sdf", "mae", "smi", "csv", "txt", "tsv", "tdt"};
```

Comment:
I think these two variables should be defined at the *class* level, not within this function. That helps document quite clearly what is supported.